.github/workflows/ci-complete.yml

```diff
@@ -118,3 +118,16 @@ jobs:
           description: 'The build scan was successfully published'
           context: 'Gradle Build Scan / Java ${{ matrix.java }}'
           state: 'success'
+
+  update-cached-tag:
+    # Only run this job if the previous workflow run was successful
+    if: github.event.workflow_run.conclusion == 'success'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Update trunk-cached branch with trunk
+        run: |
+          git switch trunk-cached || git switch -c trunk-cached
```

Comment:
Do we need to fetch trunk-cached first? Or does the checkout step fetch everything?

Reply:
As we are doing a `reset` here, it is not needed to do `fetch`. Even if we do `fetch`, we will not considering the fetched trunk-cached branch will be reset to `github.sha` as well as unfetched trunk-cached branch.

I have did two tests for this. You can check them [here](https://github.com/fonsdant/github-actions/actions/runs/10968527602/job/30459981606#step:3:11) and [here](https://github.com/fonsdant/github-actions/actions/runs/10968538249/job/30460008394#step:3:11).

This is the `git log` result:

```
$ git log --oneline 
e8da285 (HEAD -> trunk, origin/trunk-cached, origin/trunk) update ble.txt
d9c105e minimal impl
425d23e replace sha with origin trunk branch
...
```

Reply:
Thanks for the explanation. Looks like we're always doing `git switch -c` to create a local branch, so maybe we don't need the first `git switch` ?

Reply:
You are right! I will remove the `git switch`. Thanks!!